nixos/modules/services/misc/geoipupdate.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.geoipupdate;
+  configFile = pkgs.writeText "geoipupdate.conf" ''
+    AccountID ${cfg.accountID}
+    LicenseKey ${cfg.licenseKey}
+    EditionIDs ${concatStringsSep " " cfg.editionIDs}
+    ${concatStringsSep "\n" (mapAttrsToList (k: v: "${k} ${v}") cfg.extraConfig)}
+  '';
+in {
+  options.services.geoipupdate = with types; {
+    enable = mkEnableOption "the automated GeoIP updater";
+
+    accountID = mkOption {
+      description = "Account ID of your MaxMind account (or 0 for free GeoLite DBs)";
+      type = str;
+      default = "0";
+    };
+
+    licenseKey = mkOption {
+      description = "License key of your MaxMind account (or 000000000000 for free GeoLite DBs)";
+      type = str;
+      default = "000000000000";
+    };
+
+    editionIDs = mkOption {
+      description = "Edition IDs to update";
+      type = listOf str;
+      default = [ "GeoLite2-Country" "GeoLite2-City" ];
+    };
+
+    extraConfig = mkOption {
+      description = "Extra configuration to append to the configuration file";
+      type = attrsOf str;
```

Comment:
Instead of an `extraConfig` I suggest the approach in NixOS/rfcs#42: Have a `settings` option (of the same type as this one) where *all* settings are defined, including `AccountID` and such. These can then be defaulted in the config section below, e.g. `services.geoipupdate.settings.AccountID = mkDefault cfg.accountID`